File: HW 8 (Networking)/multithreading/src/main/java/com/example/multithreading/database/EnumTypeConverter.java
package com.example.multithreading.database;

import androidx.room.TypeConverter;

public class EnumTypeConverter {

    @TypeConverter
    public String enumToString(CONTACT_TYPE contactType){
        return contactType.name();
    }

    @TypeConverter
    public CONTACT_TYPE stringToEnum(String contactType){
        return CONTACT_TYPE.valueOf(contactType);
    }
}

File: HW 8 (Networking) — копия/kotlin_room/src/main/java/com/example/kotlin_room/database/EnumTypeConverter.kt
package com.example.kotlin_room.database

import androidx.room.TypeConverter

class EnumTypeConverter {
    @TypeConverter
    fun enumToString(contactType: CONTACT_TYPE): String {
        return contactType.name
    }

    @TypeConverter
    fun stringToEnum(contactType: String): CONTACT_TYPE {
        return CONTACT_TYPE.valueOf(contactType)
    }
}